### PR TITLE
[Reqwest example]: Include headers into response

### PR DESCRIPTION
**Pull request by Valerioageno**
## Topic 

Example fix 

## Motivation

While working on a reverse proxy with reqwest I think I noticed that the headers are not actually passed from the request to the response (I'm new to axum so I might be wrong).

Shouldn't be like this?

## Solution

We update the `response_builder`'s `HeadersMap`

### CODE REVIEW

File: examples/reqwest-response/src/main.rs
Comment (mladedav) [inline examples/reqwest-response/src/main.rs:71]: Nit, if you're going to iterate like this, it's more idiomatic to use a simple `for`. The docs for `for_each` itself mention it too.

Comment (Valerioageno) [reply examples/reqwest-response/src/main.rs]: Nice catch. Thanks. Updated

**Review by mladedav — APPROVED** [review]
Once #2843 lands the CI will hopefully pass. LGTM

**Comment by Valerioageno** [comment]
Hey @mladedav, thanks for your approval. I think I need one more since the rebase

**Review by mladedav — APPROVED** [review]
I'm not really an approver, so although I can click approve, it doesn't really have any value other than informational that _I_ think it's ok and that _someone_ has seen it. But merging will have to wait until an actual maintainer gets to it.

**Review by jplatte — COMMENTED** [review]
I also think this was broken before, but instead of using a completely different approach (that I think handles mutliple header values for the same header name differently), could you just add `*response_builder.headers_mut().unwrap() = headers;` to after the `extend` call that was already there, and keep the `HeaderMap::with_capacity` creation and such?

**Comment by Valerioageno** [comment]
I haven't thought about it! Thanks for spotting. Actually cleaner solution while keeping the `HeaderMap::with_capacity` initialization.

File: examples/reqwest-response/src/main.rs
Comment (jplatte) [inline examples/reqwest-response/src/main.rs:67]:
```suggestion
```

**Review by jplatte — APPROVED** [review]
Thanks!

**Pull request merged**
